### PR TITLE
refactor: rewrite GenReifiedSourceLocationClass to use bytecode dsl

### PR DESCRIPTION
**Pull request by JonathanStarup**
(related to #2591).

### CODE REVIEW

File: main/src/ca/uwaterloo/flix/language/phase/jvm/GenReifiedSourceLocationClass.scala [hunk 1]
```diff
@@ -17,229 +17,170 @@
 package ca.uwaterloo.flix.language.phase.jvm
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/language/phase/jvm/GenReifiedSourceLocationClass.scala:17]: Can you test that this generates correct code? I think if you trigger an exception then the source location should be printed.

Comment (JonathanStarup) [reply main/src/ca/uwaterloo/flix/language/phase/jvm/GenReifiedSourceLocationClass.scala:17]: `Exception in thread "main" dev.flix.runtime.MatchError: Non-exhaustive match at examples\misc\test.flix:4:5`

**Comment by JonathanStarup** [comment]
```java
public final class ReifiedSourceLocation {
    public final String source;
    public final int beginLine;
    public final int beginCol;
    public final int endLine;
    public final int endCol;

    public ReifiedSourceLocation(String var1, int var2, int var3, int var4, int var5) {
        this.source = var1;
        this.beginLine = var2;
        this.beginCol = var3;
        this.endLine = var4;
        this.endCol = var5;
    }

    public boolean equals(Object var1) {
        if (this == var1) {
            return true;
        } else if (var1 == null) {
            return false;
        } else if (this.getClass() != var1.getClass()) {
            return false;
        } else {
            ReifiedSourceLocation var2 = (ReifiedSourceLocation)var1;
            if (this.beginLine != var2.beginLine) {
                return false;
            } else if (this.beginCol != var2.beginCol) {
                return false;
            } else if (this.endLine != var2.endLine) {
                return false;
            } else {
                return this.endCol != var2.endCol ? false : Objects.equals(this.source, var2.source);
            }
        }
    }

    public int hashCode() {
        return Objects.hash(new Object[]{this.source, this.beginLine, this.beginCol, this.endLine, this.endCol});
    }

    public String toString() {
        return this.source + ":" + this.beginLine + ":" + this.beginCol;
    }
}
```

File: main/src/ca/uwaterloo/flix/language/phase/jvm/GenReifiedSourceLocationClass.scala (hunk 1 above)
Comment (JonathanStarup) [reply main/src/ca/uwaterloo/flix/language/phase/jvm/GenReifiedSourceLocationClass.scala:17]: like some of the other cases i don't think equals/hashcode is used

**Comment by magnus-madsen** [comment]
If I remember we only included equals and hashCode because we liked the DSL code?

Do we still do that? Otherwise can remove it. (We could also leave it, if it is a nice example for an eventual paper?)

**Comment by JonathanStarup** [comment]
> If I remember we only included equals and hashCode because we liked the DSL code?
> 
> Do we still do that? Otherwise can remove it. (We could also leave it, if it is a nice example for an eventual paper?)

I vote keep it, I think we should do a redundancy/optimization pass after all the refactoring

**Comment by magnus-madsen** [comment]
Thanks!

**Pull request merged**